angular/src/app/app.component.ts

```diff
@@ -0,0 +1,226 @@
+import { Component, Renderer2, NgZone } from '@angular/core';
+import { RouterOutlet } from '@angular/router';
+import { environment } from '../environments/environment.development';
+
+declare const Helpshift: any;
+declare const helpshiftConfig: any;
+
+@Component({
+  selector: 'app-root',
+  standalone: true,
+  imports: [RouterOutlet],
+  templateUrl: './app.component.html',
+  styleUrl: './app.component.css',
+})
+export class AppComponent {
+  title = 'Web Chat';
+
+  constructor(private renderer: Renderer2, private ngZone: NgZone) {}
+
+  isLoggedIn: boolean = false;
+  selectedPrivacyOption: string = 'disable';
+  selectedLauncherOption: string = 'showLauncher';
+  position: string = 'bottom-right';
+  selectedFullScreenOption: string = 'exitFullScreen';
+  unreadMessageCount: any = 0;
+  messageEventListenerIsAdded: boolean = false;
+  message: string = '';
+  messageAddEventHandlerReference: any;
+
+  newUnreadMessagesEventHandler: any;
+
+  // @NOTE: In order to add an event to the window, you should add it
+  // when the component mounts using ngOnInit()
+  // and remove it when the component unmounts using ngOnDestroy();
+
+  ngOnInit() {
+    this.loadScript();
+
+    // @NOTE: In order to add New Unread Messages event handler, add the following code.
+    this.newUnreadMessagesEventHandler = (data: any) => {
+      this.ngZone.run(() => {
+        this.unreadMessageCount = data.unreadCount;
+      });
+    };
+    Helpshift(
+      'addEventListener',
+      'newUnreadMessages',
+      this.newUnreadMessagesEventHandler
+    );
+  }
+
+  // @NOTE: In order to remove New Unread Messages event handler, add the following code.
+  ngOnDestroy() {
+    Helpshift(
+      'removeEventListener',
+      'newUnreadMessages',
+      this.newUnreadMessagesEventHandler
+    );
+  }
+
+  private loadScript() {
+    const script = this.renderer.createElement('script');
+    script.type = 'text/javascript';
+    script.text = `
+    (function () {
+
+      //  @NOTE: In order to load Web Chat, you have to set the PLATFORM_ID and DOMAIN in the script.
+      // https://developers.helpshift.com/web-chat/getting-started/#manual-embed
+
+      var PLATFORM_ID = "${environment.PLATFORM_ID}",
+        DOMAIN = "${environment.DOMAIN}",
+        LANGUAGE = "en";
+
+        // Do not modify the code below
+
+        // @NOTE: It's a good idea to set all widget options and information here.
+        // However, in the case where some information needs to be added later (async),
+        // use the updateHelpshiftConfig API https://developers.helpshift.com/web-chat/api/
+
+      window.helpshiftConfig = {
```

Comment:
Since this object does not have anything specific to TS, can we move it to a utility file ?

Reply:
Yes, we can do that 
but for now, we have made it simple basically to tell the clients where it should be placed.

Reply:
ok 👍